File: copilot-demo-labs/use-cases/02-refactoring-performance/src/ReportProcessor.ts
interface OrderItem {
  qty: number;
  price: number;
}

interface Order {
  country?: string;
  items: OrderItem[];
}

export function runLabVariant(payload: { orders: Order[] }): Record<string, number> {
  const orders = payload.orders || [];
  const totals: Record<string, number> = {};
  for (let i = 0; i < orders.length; i++) {
    const order = orders[i];
    const country = order.country || "UNKNOWN";
    if (!totals.hasOwnProperty(country)) {
      totals[country] = 0;
    }
    for (let j = 0; j < order.items.length; j++) {
      const item = order.items[j];
      totals[country] += item.qty * item.price;
    }
  }
  // Second pass for tax logic
  Object.keys(totals).forEach(c => {
    if (c === "IT") totals[c] = totals[c] * 1.22;
    else if (c === "DE") totals[c] = totals[c] * 1.19;
    else totals[c] = totals[c] * 1.10;
  });
  return totals;
}

File: copilot-demo-labs/use-cases/02-refactoring-performance/src/ReportProcessor.py
def run_lab_variant(payload: dict) -> dict:
    orders = payload.get("orders", [])
    totals = {}
    for order in orders:
        country = order.get("country") or "UNKNOWN"
        if country not in totals:
            totals[country] = 0
        for item in order.get("items", []):
            totals[country] += item["qty"] * item["price"]
    # Second pass for tax logic
    for c in list(totals.keys()):
        if c == "IT":
            totals[c] = totals[c] * 1.22
        elif c == "DE":
            totals[c] = totals[c] * 1.19
        else:
            totals[c] = totals[c] * 1.10
    return totals
